src/defined.ts

```diff
@@ -0,0 +1,37 @@
+import {ascending, descending, type Primitive} from "d3";
+
+export function defined(x: Primitive | undefined): boolean {
+  return x != null && !Number.isNaN(x);
+}
+
+export function ascendingDefined(a: Primitive | undefined, b: Primitive | undefined): number {
+  return +defined(b) - +defined(a) || ascending(a, b);
+}
+
+export function descendingDefined(a: Primitive | undefined, b: Primitive | undefined): number {
+  return +defined(b) - +defined(a) || descending(a, b);
+}
+
+export function nonempty(x: unknown): boolean {
+  return x != null && `${x}` !== "";
+}
+
+export function finite(x: number): number {
+  return isFinite(x) ? x : NaN;
+}
+
+export function positive(x: number): number {
+  return x > 0 && isFinite(x) ? x : NaN;
+}
+
+export function negative(x: number): number {
+  return x < 0 && isFinite(x) ? x : NaN;
+}
+
+export function firstof(...values: (Primitive | undefined)[]): Primitive | undefined {
```

Comment:
```suggestion
export function firstof(...values: (Array | undefined)[]): Array | undefined {
```

The only reference to firstof is https://github.com/observablehq/plot/blob/8757c028e12ba434ff2f199f8789e338140170fd/src/transforms/group.js#L229

The arguments are either Array, typedArray, or undefined. Does Array include typed arrays?

Reply:
some googling reveals that people want to define 
~~~ts
type TypedArray =
        | Int8Array
        | Int16Array
        | Int32Array
        | Uint8Array
        | Uint8ClampedArray
        | Uint16Array
        | Uint32Array
        | Float32Array
        | Float64Array
~~~

is this something we want to do in a centralized extra.d.ts file?